playbooks/roles/check_duplicate_irlm/tasks/main.yml

```diff
@@ -0,0 +1,55 @@
+---
+#
+# Copyright 2024 IBM Inc. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Check if IRLM is already up
+- name: Check for duplicate IRLM
+  collections:
+      - ibm.ibm_zos_core
+      - ibm.ibm_zos_ims
+  block:
+    - name: Query IRLM subsystem
+      ibm.ibm_zos_core.zos_operator:
+        cmd: 'D SSI,SUB={{ DXR_IRLM_SSID }}'
+      register: irlm_output
+
+    - name: Check if IRLM subsystem exists 
+      ansible.builtin.set_fact:
+        irlm_is_up: true
+      when:
+        - irlm_output.content is defined
+        - '"THERE IS NO INFORMATION TO DISPLAY" not in irlm_output.content | join'
+
+    - name: 'IRLM subsystem {{ DXR_IRLM_SSID }} already exists. Trying to reconcile...'
+      ansible.builtin.debug:
+        msg: 'IRLM subsystem {{ DXR_IRLM_SSID }} already exists. Trying to reconcile...'
+      when:
+        - reconcile_run
+        - irlm_is_up
+        - tmdb_cr_instance.resources[0].status.run_result != 'failedInputValidation'
+
+    # Reconcile run and failedInputValidation, fail it
+    - name: 'Duplicate IRLM subsystem {{ DXR_IRLM_SSID }}. Please use a different SSID for the IRLM subsystem.'
+      ansible.builtin.fail:
+        msg: 'Duplicate IRLM subsystem {{ DXR_IRLM_SSID }}. Please use a different SSID for the IRLM subsystem.'
+      when: 
+        - reconcile_run
+        - irlm_is_up
+        - tmdb_cr_instance.resources[0].status.run_result == 'failedInputValidation'
+
+    # IRLM already up and a new 'Create' comes in, fail it with 'failedInputValidation' status
+    - block: 
+      - name: Update CR instance run status
```

Comment:
Ahh, I always forget about this with the acronyms! 
Btw, I'm reusing a lot of this code from the other roles, so I didn't put much thought on these names to keep it consistent with what's already there. It would be good to update the task names on the other roles as well. 
Just did a quick search and the same task name is repeated on other 16 places.